core/layouts/geary/layout--geary.tpl.php

```diff
@@ -31,82 +31,92 @@
-      <div class="container container-fluid">
-        <div class="row">
-          <?php print $content['header']; ?>
-        </div><!-- /.row -->
+      <div class="l-header-inner container container-fluid">
+        <?php print $content['header']; ?>
       </div><!-- /.container -->
     </header>
   <?php endif; ?>
 
-  <div class="l-wrapper">
-    <div class="l-wrapper-inner container container-fluid">
+  <section class="l-wrapper">
+    <div class="l-wrapper-inner">
 
       <?php if ($messages): ?>
-        <div class="l-messages row" role="status" aria-label="<?php print t('Status messages'); ?>">
-          <div class="col-xs-12">
+        <div class="l-messages" role="status" aria-label="<?php print t('Status messages'); ?>">
+          <div class="l-messages-inner container container-fluid">
             <?php print $messages; ?>
           </div>
         </div>
       <?php endif; ?>
 
-      <div class="l-page-header row">
-        <a id="main-content"></a>
-        <?php print render($title_prefix); ?>
-        <?php if ($title): ?>
-          <div class="col-xs-12">
+      <header class="l-page-header">
+        <div class="l-page-header-inner container container-fluid">
+          <a id="main-content"></a>
+          <?php print render($title_prefix); ?>
+          <?php if ($title): ?>
             <h1 class="title" id="page-title"><?php print $title; ?></h1>
-          </div>
-        <?php endif; ?>
-        <?php print render($title_suffix); ?>
-      </div>
+          <?php endif; ?>
+          <?php print render($title_suffix); ?>
+        </div>
+      </header>
 
       <?php if ($tabs): ?>
-        <nav class="tabs row" role="tablist" aria-label="<?php print t('Admin Content Navigation Tabs'); ?>">
+        <nav class="tabs container container-fluid" role="tablist" aria-label="<?php print t('Admin Content Navigation Tabs'); ?>">
           <?php print $tabs; ?>
         </nav>
       <?php endif; ?>
 
       <?php print $action_links; ?>
 
       <?php if (!empty($content['top'])): ?>
-        <div class="row">
-          <div class="l-top col-md-12">
+        <div class="l-top">
+          <div class="l-top-inner container container-fluid">
             <?php print $content['top']; ?>
           </div>
         </div>
       <?php endif; ?>
 
-      <div class="l-thirds row">
-        <div class="col-md-4">
-          <?php print $content['third1']; ?>
-        </div>
-        <div class="col-md-4">
-          <?php print $content['third2']; ?>
-        </div>
-        <div class="col-md-4">
-          <?php print $content['third3']; ?>
+      <?php if (!empty($content['third1']) || !empty($content['third2']) || !empty($content['third3'])): ?>
+        <div class="l-thirds">
+          <div class="l-thirds-inner container container-fluid">
+            <div class="l-thirds-inner-2 row">
+              <div class="l-thirds-region col-md-4">
```

Comment:
On top of that, flexbox might make elements grow in width if they're missing siblings.